File: android/app/src/main/java/noah/moohvie/models/ShopItem.kt
package noah.moohvie.models

import androidx.compose.ui.graphics.Color

enum class ShopItemCategory {
    THEME,
    TITLE,
    BADGE;

    val label: String
        get() = when (this) {
            THEME -> "Thèmes"
            TITLE -> "Titres"
            BADGE -> "Badges"
        }
}

data class ShopItem(
    val id: String,
    val category: ShopItemCategory,
    val name: String,
    val cost: Int,
    /** Couleur d'accent pour les items de type THEME. */
    val accentColor: Color? = null,
    /** Emoji utilisé pour les items de type BADGE. */
    val badgeIcon: String? = null,
)

File: Moohvie/Models/shopitem.swift
import SwiftUI

enum ShopItemCategory: String, CaseIterable {
    case theme
    case title
    case badge

    var label: String {
        switch self {
        case .theme: return "Thèmes"
        case .title: return "Titres"
        case .badge: return "Badges"
        }
    }
}

struct ShopItem: Identifiable, Hashable {
    let id: String
    let category: ShopItemCategory
    let name: String
    let cost: Int

    /// Couleur d'accent pour les items de type `.theme`.
    var accentColor: Color?
    /// SF Symbol pour les items de type `.badge`.
    var badgeIcon: String?
}
